### PR TITLE
Clean some deprecated classes/methods usages in htsjdk

### LINKED ISSUE #330: Remove deprecated methods in SequenceUtil
SequenceUtil contains deprecated versions of the following methods:
- countMismatches
- sumQualitiesOfMismatches
- calculateSamNmTag

The main difference is that `referenceBases` is passed in as a `char[]` in the deprecated versions, rather than a `byte[]`. Due to the structure of the code, it is not easy to factor out the internal logic without a performance hit, so the code is mostly duplicated (and as a result, the deprecated implementations are out-of-date).

Apparently GATK used to pass in a char array, but has since been updated, so removing the deprecated versions will not affect GATK.

Would anyone else object to removing the deprecated methods?

### PR DESCRIPTION
**Pull request by magicDGS**
### Description

I explore some of the deprecated classes/methods when cleaning some of my own code, and I found that there are usages of some of this methods in other parts of htsjdk. This is a PR to clean the usages and to ask for removal of some deprecated classes/methods no longer used. This includes:
- ~~Fixes #330~~ Already fixed in other PR
- Changed all `Feature.getChr()` to `Feature.getContig()` usages in htsjdk.
- Remove `SRAAccession.isSupported()` usages in the tests and change for the recomended one.
- Annotating some deprecated methods with `@Deprecated` and `/** @deprecated */`
- Adding javadoc `{@link }` for pointing to new/correct usage.
- ~~Asking for removal of classes and methods for different packages, including gelitext support, `VariantContextWriterFactory` and others.~~ Edited: open a PR for removal and discussion.

I divided the PR in several commits according to common changes, to make it easier to review. I think that this will help in the maintenance of the htsjdk library. If the removal of some classes is made effective, this will break compatibility with previous versions.

### Checklist
- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [x] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by magicDGS** [comment]
Could you have a look, @lbergelson? I'm pinging you because you were working on deprecated classes/methods in #698 and #699.

**Comment by magicDGS** [comment]
Friendly ping @lbergelson!

File: src/main/java/htsjdk/samtools/SamPairUtil.java [hunk 1]
```diff
@@ -322,11 +334,13 @@ public static void setMateInformationOnSupplementalAlignment( final SAMRecord su
 
     /**
      * This method will clear any mate cigar already present.
+     * @deprecated use {@link #setProperPairAndMateInfo(SAMRecord, SAMRecord, List)} instead
```
Comment (magicDGS) [inline src/main/java/htsjdk/samtools/SamPairUtil.java]: I changed the body.

File: src/main/java/htsjdk/samtools/SamPairUtil.java [hunk 2]
```diff
@@ -335,12 +349,34 @@ public static void setProperPairAndMateInfo(final SAMRecord rec1, final SAMRecor
...
+     */
+    public static void setProperPairAndMateInfo(final SAMRecord rec1, final SAMRecord rec2,
+                                                final List<PairOrientation> exepectedOrientations) {
+        setProperPairAndMateInfo(rec1, rec2, exepectedOrientations, false);
+    }
+
+    /**
+     * @param rec1
+     * @param rec2
+     * @param exepectedOrientations
```
Comment (magicDGS) [inline src/main/java/htsjdk/samtools/SamPairUtil.java]: params without description removed.

File: src/main/java/htsjdk/tribble/example/CountRecords.java [hunk 3]
```diff
@@ -193,10 +193,13 @@ public static FeatureCodec getFeatureCodec(File featureFile) {
         //    return new VCFCodec();
         if (featureFile.getName().endsWith(".bed") || featureFile.getName().endsWith(".BED") )
             return new BEDCodec();
+        // TODO: can this commented lines be removed?
```
Comment (magicDGS) [inline src/main/java/htsjdk/tribble/example/CountRecords.java]: Done!

**Comment by magicDGS** [comment]
I agree with you @lbergelson, we should separate removal from this PR (I didn't remove anything here in my previous commits). I open several PRs for removing deprecated code, and I let here only deprecated javadoc/annotations, and removal of deprecated code inside HTSJDK (in addition to your 👍).

Back to you for more comments!

**Comment by lbergelson** [comment]
@magicDGS 👍 Thanks for the cleanup.

**Pull request merged**